packages/firestore/src/local/encoded_resource_path.ts

```diff
@@ -175,25 +175,3 @@ export function decode(path: EncodedResourcePath): ResourcePath {
 
   return new ResourcePath(segments);
 }
-
-/**
- * Computes the prefix successor of the given path, computed by encode above.
- * A prefix successor is the first key that cannot be prefixed by the given
- * path. It's useful for defining the end of a prefix scan such that all keys
- * in the scan have the same prefix.
- *
- * Note that this is not a general prefix successor implementation, which is
- * tricky to get right with Strings, given that they encode down to UTF-8.
- * Instead this relies on the fact that all paths encoded by this class are
- * always terminated with a separator, and so a successor can always be
- * cheaply computed by incrementing the last character of the path.
- */
-export function prefixSuccessor(
```

Comment:
Just for my own information, is it acceptable to delete exported functions as long as they are not "publicly exported" in https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore-types/index.d.ts?  Otherwise, wouldn't this be a backwards-incompatible change?

Reply:
In general, if code is not in the api/ folder, it can't be used by our users.

In more technical terms, only code that is exported from our index.ts files is considered part of the public API:
https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/register-module.ts

Furthermore, the files under /api should match our types  (https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore-types/index.d.ts). All code in the exported classes above should either be part of the public API or prefixed with an underscore. We don't always get this right though - see https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/api/database.ts#L653. `logLevel` is not in our types, but can be accessed since it is exported as part of Firestore.

Reply:
The Tree-Shaking in the browser builds doesn't actually work very well. I am trying to fix that (currently not working though - https://github.com/firebase/firebase-js-sdk/pull/2829), but for the "older builds" only excludes code that in modules that aren't loaded at all. Only the ESM2017 build will not include functions, but also only if they are deemed "side-effect free".